mantidproject/mantid
# Project Recovery: Handle multiple instances of mantid properly

**Description of work.**
Added the ability for project recovery to properly handle recovering multiple instances of mantid. For example if I am running instances 1 and 2, and 1 crashes, then I launch a 3rd instance I should be able to recover 1s or if both crash I should be able to recover from both, it will do it in order of which was made most recently, so which ever was edited last.

**To test:**
Make sure you launch this version of Mantid and not different versions as it cannot detect different versions.

1. Launch 2 instances of mantid
2. Both instances should have a loaded workspace and wait for project recovery to save itself
3. Crash the first instance
4. Wait for the second instance to save (To prove it doesn't save over it)
5. Relaunch mantid
6. Click yes on recovery and see that it is the same workspace that you had on the first instance

Fixes #23494  

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
